### PR TITLE
feat: support for buildEnvironmentVariables

### PR DESCRIPTION
**Pull request by yuhsak**
Add support for `buildEnvironmentVariables`.

Currently only `environmentVariables` feature is supported with `env_vars` and `env_vars_file` options.

This PR adds `build_env_vars` and `build_env_vars_file` options in the same way.

### CODE REVIEW

File: src/cloudFunction.ts [hunk 1]
```diff
@@ -197,13 +222,16 @@ export class CloudFunction {
    * @param envVarsFile env var file path.
    * @returns map of type {KEY1:VALUE1}
    */
-  protected parseEnvVarsFile(envVarFilePath: string): KVPair {
+  protected parseEnvVarsFile(
+    envVarFilePath: string,
+    optionName = 'env_vars_file',
```
Comment (bharathkkb) [inline src/cloudFunction.ts:227]: nit: I think we just use the base name instead like `myvars.yaml must contain only key/value pair of strings. ...`

**Review by bharathkkb — COMMENTED** [review]
Thanks for adding this!

File: src/cloudFunction.ts (hunk 1 above)
Comment (yuhsak) [reply src/cloudFunction.ts:227]: Hmm that's true! I'll fix it.

**Comment by yuhsak** [comment]
@bharathkkb 
Sorry for replying late. Please review when you'd have a time!

**Comment by bharathkkb** [comment]
@yuhsak looks like some of the tests using the envvarfile may need updating since we now output the filename

**Comment by yuhsak** [comment]
@bharathkkb 
Oops sorry. I just updated that tests.

**Pull request closed without merging**